src/prefect/cli/worker.py

```diff
@@ -156,5 +158,11 @@ async def start(
 
             started_event = await worker._emit_worker_started_event()
 
+            # start healthcheck ASGI server
+            # in a background thread
+            server_thread = threading.Thread(target=run_healthcheck_server)
+            server_thread.daemon = True
+            server_thread.start()
```

Comment:
You can set `daemon` in the `Thread(...)` call. You should also give it a clear name so we know what the thread is when debugging. 

Running it in a worker thread should prevent it from taking over any signal handling, which is kind of nice? It definitely adds complexity though and probably isn't the best for performance. You can run it in the background on the event loop instead of a thread by using `tg.start_soon`.

Reply:
damn, throwing it into an async task group re-introduces the inability to SIGINT the process.  there's a hacky workaround to using `tg`s by invoking `exit()` from a custom SIGINT handler on the fastapi instance using `signals`, but im not sure of the side effects.  might be ok since the handler only exists on the webserver, but worth discussing

Reply:
We could grab the signal handler that's registered then start uvicorn then set a new signal handler which calls the stashed signal handler and uvicorn's. Kind of a pain but might be worth it? I could also probably try to get this fixed upstream.

Reply:
so something like this?
https://github.com/PrefectHQ/prefect/pull/9687/files#diff-eb8a03a8ac41aec52fe38a471a96a468ffba04db74fb97149c65f9ee1efd2266R9-R21

though something about this making the unit tests unhappy

Reply:
>though something about this making the unit tests unhappy

looks like select tests are timing out, one example is:

```
========================================================================================================================== short test summary info ===========================================================================================================================
FAILED tests/events/instrumentation/test_events_workers_instrumentation.py::test_lifecycle_events - Failed: Timeout >60.0s
================================================================================================================== 1 failed, 81 passed in 77.46s (0:01:17) ===================================================================================================================
```